### PR TITLE
DI-241 Create DynamoDB Table

### PR DESCRIPTION
**Pull request by rohith1122**
## Link to JIRA Ticket

- https://nhsd-jira.digital.nhs.uk/browse/DI-241

## Description

Created Dynamodb table using Terraform for storing change event 
Implemented a function to add change events to the DynamoDB Table

### Noteworthy Changes

- Added moto library as a dev dependency for unit tests ( to test AWS Services like Dynamodb, SQS, etc.)

## Type of change
- New feature (non-breaking change which adds functionality)

## Testing

Please tick the testing that has been completed

- [x] Unit tests
- [ ] Component tests

## Developer Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the [code formatting checks](../README.md#code-quality)
- [x] I have run the [code quality checks](../README.md#code-quality)
- [x] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [x] Unit test code coverage is at or above 80%
- [x] New and existing unit tests pass locally with my changes
- [x] Tests have added that prove my fix is effective or that my feature works (Component tests)
- [x] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

File: application/event_processor/dynamodb.py
Comment (JackPlowman) [inline application/event_processor/dynamodb.py:14]: Can this file be removed to the common folder so it can be shared with the new fifo-dlq-handler.

File: infrastructure/stacks/lambda-iam-roles/main.tf
Comment (JackPlowman) [inline infrastructure/stacks/lambda-iam-roles/main.tf:57]: Can these new permissions be moved to their own statement and locked to the specific dynamodb table with this variable `TF_VAR_change_events_table_name`

**Review by JackPlowman — CHANGES_REQUESTED** [review]
Overall looks great, have checked with DI-242 branch which is based off this one. Couple of small changes, one to improve shared code between lambdas and other to improve access control

File: application/event_processor/dynamodb.py
Comment (rohith1122) [reply application/event_processor/dynamodb.py:14]: Sure. It makes sense

File: infrastructure/stacks/lambda-iam-roles/main.tf
Comment (rohith1122) [reply infrastructure/stacks/lambda-iam-roles/main.tf:57]: Sure

File: application/event_processor/event_processor.py
Comment (rohith1122) [reply application/event_processor/event_processor.py:138]: I do not know about this but for some reason, Matt has removed these lines .I will add these lines back

**Review by ed-sparkes — APPROVED** [review]

**Review by JackPlowman — APPROVED** [review]
LGTM 👍 , PLease don't merge until testers have given their approval

**Review by ed-sparkes — APPROVED** [review]

**Review by JackPlowman — APPROVED** [review]

**Pull request merged**
